src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/ImageResourceConverter.shared.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.Forms;
+#if NETSTANDARD1_0 || UAP10_0
+using System.Reflection;
+#endif
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	/// Converts embedded image resource ID to it ImageSource.
+	/// </summary>
+	public class ImageResourceConverter : IValueConverter
+	{
+		/// <summary>
+		/// Converts embedded image resource ID to it ImageSource.
+		/// </summary>
+		/// <param name="value">The value to convert.</param>
+		/// <param name="targetType">The type of the binding target property. This is not implemented.</param>
+		/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
+		/// <param name="culture">The culture to use in the converter. This is not implemented.</param>
+		/// <returns>The ImageSource related to the provided resource ID of the embedded image. If it's null it will returns null.</returns>
+		public object? Convert(object value, Type targetType, object parameter, CultureInfo culture)
+		{
+			var imageId = (string)value;
```

Comment:
Can we avoid explicit cast here? It can cause cast-exception here.

Reply:
Do you suggest replacing it with the "as" operator ?